h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/UpdateAuxTreeWeightsTask.java

```diff
@@ -0,0 +1,85 @@
+package hex.tree.xgboost.predict;
+
+import biz.k11i.xgboost.Predictor;
+import biz.k11i.xgboost.gbm.GBTree;
+import biz.k11i.xgboost.tree.RegTree;
+import biz.k11i.xgboost.tree.RegTreeNode;
+import hex.DataInfo;
+import hex.tree.xgboost.XGBoostModelInfo;
+import hex.tree.xgboost.XGBoostOutput;
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.NewChunk;
+import water.util.ArrayUtils;
+
+public class UpdateAuxTreeWeightsTask extends MRTask<UpdateAuxTreeWeightsTask> {
+
+    // IN
+    private final Predictor _p;
+    private final DataInfo _di;
+    private final boolean _sparse;
+    // OUT
+    private final double[/*treeId*/][/*leafNodeId*/] _nodeWeights;
+
+    public UpdateAuxTreeWeightsTask(DataInfo di, XGBoostModelInfo modelInfo, XGBoostOutput output) {
+        _p = PredictorFactory.makePredictor(modelInfo._boosterBytes, null, false);
+        _di = di;
+        _sparse = output._sparse;
+
+        if (_p.getNumClass() > 2) {
+            throw new UnsupportedOperationException("Calculating contributions is currently not supported for multinomial models.");
```

Comment:
Is seems to me that it has nothing common with contributions

Reply:
copy & paster error, thanks for catching it - fixed